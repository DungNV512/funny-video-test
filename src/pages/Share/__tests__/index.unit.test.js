import React from 'react';
import { render, screen } from '@testing-library/react';
import { ShareModule } from '..';
import '@testing-library/jest-dom'

describe('ShareModule', () => {
  it('matches snapshot', () => {
    const { container } = render(<ShareModule />);
    // eslint-disable-next-line testing-library/no-container
    expect(container.cloneNode(true)).toMatchSnapshot();
  });

  it('renders default state', () => {
    render(<ShareModule />)

    // signed in
    const loggedInText = screen.getByText('Welcome');
    expect(loggedInText).toBeInTheDocument();
    const emailAddressText = screen.getByText('test@email.com');
    expect(emailAddressText).toBeInTheDocument();

    const buttonShare = screen.getByRole('button', {name: /share a movie/i})
    expect(buttonShare).not.toBeDisabled()
  })
});
