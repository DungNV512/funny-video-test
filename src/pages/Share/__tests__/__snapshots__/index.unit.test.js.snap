// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareModule matches snapshot 1`] = `
<div>
  <div>
    <span>
      Welcome 
      <b>
        test@email.com
      </b>
    </span>
    <button>
      Share a movie
    </button>
    <button>
      Logout
    </button>
  </div>
  <form
    autocomplete="off"
    novalidate=""
  >
    <label
      for="url"
    >
      Youtube URL:
      <input
        id="url"
        name="url"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Share
    </button>
  </form>
</div>
`;
